### ISSUE TITLE
OpenSSL::X509::CertificateError: nested asn1 error

### ISSUE THREAD

**Issue by sudarshna-patel** [ruby/openssl#201]
`str = "some string"`
`OpenSSL::X509::Certificate.new("-----BEGIN CERTIFICATE-----\n#{str}\n-----END CERTIFICATE-----")`

I executed in my system, it is working fine but when i tried to execute on my remote server, its throwing error OpenSSL::X509::CertificateError: nested asn1 error. I even updated my openssl version, its OpenSSL 1.0.2o

**Comment by sebavidal10** [comment]
Hi!, you can resolve it? i have the same problem :S all ok in local machine, but in production don't :/

**Comment by maxgronlund** [comment]
Did you resolve it? I got the same challenge

**Comment by Jay-Plumb** [comment]
I am having the same issue. Locally everything works well but on Heroku production I get `OpenSSL::X509::CertificateError: nested asn1 error`

**Comment by Jay-Plumb** [comment]
I just wanted to provide an explanation into the issue I faced yesterday (which has now been resolved). 

The issue I faced was that locally, the `.crt` was stored as an ENV variable which required `\n` to be manually inserted. When testing out via heroku, simply copying over what I had configured locally did not work for Heroku. Instead I had to run the command 

```shell
$ heroku config:set CRT_FILE="$(cat path/to/user.crt)" -a HEROKU_APP_NAME
```

That solved the issue for me so hopefully that helps anyone else in the future.
